prov/net/src/xnet_progress.c

```diff
@@ -1004,6 +1006,33 @@ static void xnet_progress_uring(struct xnet_progress *progress,
 	ofi_uring_cq_advance(&uring->ring, nready);
 }
 
+int xnet_uring_cancel(struct xnet_progress *progress,
+		      struct xnet_uring *uring,
+		      struct ofi_sockctx *canceled_ctx,
+		      struct ofi_sockctx *ctx)
+{
+	bool submitted = false;
+	int ret = 0;
+
+	assert(xnet_progress_locked(progress));
+	while (canceled_ctx->uring_sqe_inuse) {
+		if (!submitted) {
+			ret = ofi_sockctx_uring_cancel(uring->sockapi,
+						       canceled_ctx,
+						       ctx);
+			if (ret == -OFI_EINPROGRESS_URING) {
+				ofi_uring_submit(&uring->ring);
+				submitted = true;
+			}
+			else if (ret != -FI_EAGAIN)
```

Comment:
nit: move else to same line as } and add braces to else-if statement

Reply:
Fixed. Thanks!